meta1v2/meta1_backend.c

```diff
@@ -110,3 +121,27 @@ meta1_backend_get_prefixes(struct meta1_backend_s *m1)
 	return m1 ? m1->prefixes : NULL;
 }
 
+void
+meta1_backend_close_callback(struct sqlx_sqlite3_s *sq3,
+		struct meta1_backend_s *m1b)
+{
+	gint64 seq = 1;
+	EXTRA_ASSERT(sq3 != NULL);
+
+	if (!sq3->deleted)
+		return;
+
+	struct oio_url_s *url = oio_url_empty ();
+	oio_url_set(url, OIOURL_NS, m1b->ns_name);
```

Comment:
Namespace name is in `sq3->name->ns`. Instead of a `struct meta1_backend_s *`, the second parameter of the callback could be a `struct hc_resolver_s *`, which would avoid adding a field to `struct meta1_backend_s`.